static/app/components/compactSelect/utils.tsx

```diff
@@ -60,6 +61,88 @@ export function getDisabledOptions<Value extends React.Key>(
   }, []);
 }
 
+/**
+ * Recursively finds the option(s) that don't match the designated search string or are
+ * outside the list box's count limit.
+ */
+export function getHiddenOptions<Value extends React.Key>(
+  items: SelectOptionOrSection<Value>[],
+  search: string,
+  limit: number = Infinity
+): Set<Value> {
```

Comment:
Where the magic happens… The size limit should apply _after_ filtering using the `search` string.